src/register/data/reducers.js

```diff
@@ -39,6 +40,7 @@ export const defaultState = {
   usernameSuggestions: [],
   validationApiRateLimited: false,
   shouldBackupState: false,
+  autoGeneratedRegExpVariation: '',
```

Comment:
this doesn't convey the purpose of this flag. Can we rename it please?
```suggestion
  autoGeneratedUsernameExperimentVariation: '',
```